src/zjs_sensor.c

```diff
@@ -0,0 +1,512 @@
+// Copyright (c) 2016, Intel Corporation.
+#ifdef BUILD_MODULE_SENSOR
+#ifndef QEMU_BUILD
+#ifndef ZJS_LINUX_BUILD
+// Zephyr includes
+#include <zephyr.h>
+#endif
+
+#include <string.h>
+
+// ZJS includes
+#include "zjs_sensor.h"
+#include "zjs_callbacks.h"
+#include "zjs_ipm.h"
+#include "zjs_util.h"
+
+#define ZJS_SENSOR_TIMEOUT_TICKS                      500
+
+static struct nano_sem sensor_sem;
+
+enum sensor_state {
+    SENSOR_STATE_IDLE,
+    SENSOR_STATE_ACTIVATING,
+    SENSOR_STATE_ACTIVATED,
+    SENSOR_STATE_ERRORED,
+};
+
+typedef struct sensor_handle {
+    int32_t id;
+    enum sensor_channel channel;
+    enum sensor_state state;
+    double sensor_value[3];
```

Comment:
Fine for now, but not generic enough. See https://github.com/01org/zephyr.js/pull/353/commits/294d36b85640c752a6840df72dbe3073278b9aa7#r86344326